flow/layers/physical_shape_layer.cc

```diff
@@ -45,35 +42,33 @@ PhysicalShapeLayer::PhysicalShapeLayer(SkColor color,
     // an SkPath.
     frameRRect_ = SkRRect::MakeRect(path.getBounds());
   }
-
-  set_dimensions(frameRRect_);
 }
 
 void PhysicalShapeLayer::Preroll(PrerollContext* context,
                                  const SkMatrix& matrix) {
   TRACE_EVENT0("flutter", "PhysicalShapeLayer::Preroll");
-
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context, UsesSaveLayer());
-  PhysicalShapeLayerBase::Preroll(context, matrix);
 
-  if (elevation() == 0) {
+  context->total_elevation += elevation_;
```

Comment:
I do think it's a potential foot gun to leave it on individual layers to remember to update total_elevation_ correctly if they happen to be elevated. Right now it's only two places so it's reasonable, but this is the type of bug that wouldn't be immediately obvious if a third was introduced for some reason and didn't feed back into the global state. I don't have any immediate suggestions on how to fix it though.

Reply:
Agree with Michael on this. It might make sense for us unify the layer state management pieces. Maybe making the `MutatorStack` struct more general would be one way of addressing this.

Reply:
Yeah, I agree.

Fortunately I think context->total_elevation is going to go away in my next PR and this problem will be moot